File: backend/src/main/java/com/_blog/service/AdminService.java
package com._blog.service;

import java.util.List;
import java.util.Map;

import com._blog.entity.Role;
import com._blog.entity.User;

public interface AdminService {

    Map<String, Object> overview();

    List<Map<String, Object>> listUsers();

    User updateRole(Long userId, Role role);

    void deleteUser(Long userId);

    User banUser(Long userId, boolean banned);

    List<Map<String, Object>> listPosts();

    void deletePost(String postId);
}

File: frontend/src/app/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class AdminService {
  private apiUrl = 'http://localhost:8080/api/admin';

  constructor(private http: HttpClient) {}

  getOverview(): Observable<any> {
    return this.http.get(`${this.apiUrl}/overview`);
  }

  getUsers(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/users`);
  }

  getPosts(): Observable<any[]> {
    return this.http.get<any[]>(`${this.apiUrl}/posts`);
  }

  banUser(userId: string, banned: boolean): Observable<any> {
    return this.http.put(`${this.apiUrl}/users/${userId}/ban`, { banned });
  }

  deleteUser(userId: string): Observable<any> {
    return this.http.delete(`${this.apiUrl}/users/${userId}`);
  }

  deletePost(postId: string): Observable<any> {
    return this.http.delete(`${this.apiUrl}/posts/${postId}`);
  }
}